core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/ClusterBootstrapper.java

```diff
@@ -145,11 +152,27 @@ private void bootstrapOnHostInternal(Stack stack, Consumer<Stack> saveOrUpdateSa
             InstanceMetaData primaryGateway = stack.getPrimaryGatewayInstance();
             saveOrchestrator(stack, primaryGateway);
             checkIfAllNodesAvailable(stack, nodes, primaryGateway);
+        } catch (CloudbreakOrchestratorFailedException e) {
+            checkIfAnyInstanceIsNotInStartedState(stack, e);
+            throw new CloudbreakException(e);
         } catch (Exception e) {
             throw new CloudbreakException(e);
         }
     }
 
+    private void checkIfAnyInstanceIsNotInStartedState(Stack stack, CloudbreakOrchestratorFailedException e) throws CloudbreakException {
+        Set<InstanceMetaData> runningInstances = instanceMetaDataService.findNotTerminatedForStack(stack.getId());
+        List<CloudVmInstanceStatus> instanceStatuses = stackInstanceStatusChecker.queryInstanceStatuses(stack, runningInstances);
+        Optional<CloudVmInstanceStatus> notStartedInstance = instanceStatuses.stream()
+                .filter(instance -> !InstanceStatus.STARTED.equals(instance.getStatus()))
+                .findAny();
+        if (notStartedInstance.isPresent()) {
+            CloudInstance cloudInstance = notStartedInstance.get().getCloudInstance();
+            throw new CloudbreakException("Node " + cloudInstance.getInstanceId() + " was deleted or went missing during cluster install. " +
```

Comment:
can we print every instance which is in a bad state? instance + status for every instance